### PR TITLE
Analytics error fallback, fix opt-out

### PR DESCRIPTION
**Pull request by RAMOhio**
Fixed the analytics opt-out functionality to check if the user has opted out before using Insight library for analytics. Checks if something in the Insight library failed (i.e. on initialization) and if so, trip the no-analytics flag.

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -15,24 +15,29 @@ const packageJsonFile = 'package.json';
...
   initializing() {
     this.version = pkg.version;
-    this.insight = new Insight({trackingCode: 'UA-93069862-1', pkg});
     this.props = {};
 
+    // Try to initialize analytics. Insight is broken for some users, so if it fails, proceed as if the --no-analytics flag was present.
+    try{
+      this.insight = (!this.options.analytics || process.env.DISABLE_NGX_ANALYTICS) ? { optOut: true } : new Insight({trackingCode: 'UA-93069862-1', pkg});
+    }catch{
+      this.insight = { optOut: true };
```
Comment (sinedied) [inline generators/app/index.js]: Would be better to just noop the `track` function then, ie: `this.insight = { track: () => {} };`

File: generators/app/index.js [hunk 2]
```diff
@@ -68,7 +73,9 @@ class NgxGenerator extends Generator {
         )})\n`
       );
       this.log(`${chalk.yellow("Make sure you don't have uncommitted changes before overwriting files!")}`);
-      this.insight.track('update', fromVersion, 'to', this.version);
+      if(!this.insight.optOut){
```
Comment (sinedied) [inline generators/app/index.js]: no need for all this (and following checks) using if `insight.track` is noop'ed 😉

File: generators/app/index.js [hunk 3]
```diff
@@ -206,8 +218,6 @@ module.exports = Generator.make({
     'electron-windows': props => props.desktop && props.desktop.includes('windows'),
     'electron-mac': props => props.desktop && props.desktop.includes('mac'),
     'electron-linux': props => props.desktop && props.desktop.includes('linux'),
-    'tools-hads': props => props.tools && props.tools.includes('hads'),
-    'tools-jest': props => props.tools && props.tools.includes('jest'),
-    'tools-karma': props => props.tools && !props.tools.includes('jest')
+    'tools-hads': props => props.tools && props.tools.includes('hads')
```
Comment (sinedied) [inline generators/app/index.js]: whoops, make sure you rebase your code on the current master or this will cause a regression

**Comment by sinedied** [comment]
I think the `lint` part is missing in the contribution guide, I need to update it later my bad 🐳

**Comment by RAMOhio** [comment]
I've learned that I know even less about git through the CLI than I thought... I think I've got everything figured out though, opening another pull request here in a sec with the comments fixed. Thanks for the info, learned some new things about JS today (I still suck at JS, but hey, learning is fun lol)

**Pull request closed without merging**